WooCommerce/WooCommerceTests/ViewRelated/Orders/Order Details/OrderDetailsDataSourceTests.swift

```diff
@@ -0,0 +1,144 @@
+import Foundation
+import XCTest
+
+import Yosemite
+
+import protocol Storage.StorageManagerType
+import protocol Storage.StorageType
+
+@testable import WooCommerce
+
+/// Test cases for `OrderDetailsDataSourceTests`
+///
+final class OrderDetailsDataSourceTests: XCTestCase {
+
+    private typealias Title = OrderDetailsDataSource.Title
+
+    private var storageManager: StorageManagerType!
+
+    private var storage: StorageType {
+        storageManager.viewStorage
+    }
+
+    override func setUp() {
+        super.setUp()
+        storageManager = MockupStorageManager()
+    }
+
+    override func tearDown() {
+        storageManager = nil
+        super.tearDown()
+    }
+
+    func testPaymentSectionIsShownRightAfterTheProductsAndRefundedProductsSections() {
+        // Given
+        let order = makeOrder()
+
+        insert(refund: makeRefund(orderID: order.orderID, siteID: order.siteID))
+
+        let dataSource = OrderDetailsDataSource(order: order, storageManager: storageManager)
+        dataSource.configureResultsControllers { }
+
+        // When
+        dataSource.reloadSections()
+
+        // Then
+        let actualTitles = dataSource.sections.map(\.title)
+        let expectedTitles = [
+            nil,
+            Title.product,
+            Title.refundedProducts,
+            Title.payment,
+            Title.information,
+            Title.notes
+        ]
+
+        XCTAssertEqual(actualTitles, expectedTitles)
+    }
+
+}
+
+// MARK: - Test Data
+
+private extension OrderDetailsDataSourceTests {
+    func makeOrder() -> Order {
+        Order(siteID: 1_000,
```

Comment:
might one of the helpers in `MockOrders` simplify mock creation here?

Reply:
Thanks for the nudge. My only concern with the `MockOrders` sample data is they provide a lot of default values. It can be challenging to figure out **what data is actually being tested** in this unit test. I think a better test data would be:

1. Have **only the required properties** set to a default value.
2. Everything else is `nil`
3. Provide method arguments to change the properties to be tested.

The `copy()` pattern should work well with this. I really need to get that done soon. 🙁 

---

Anyway, I added a `makeOrder()` helper in that: 5e8ac30a5c85a61b2dcbfe0662c6d95f6938c57c. Less code is still better after all. 😄 But let me know if I'm using this class wrong!